kernel/os/src/arch/sim-armv7/os_arch_stack_frame.s

```diff
@@ -40,6 +40,7 @@ os_arch_frame_init:
     beq     end
     mov	    r1, r0
     ldr     r0, [sp, #4]
+    and     sp, sp, #0xfffffff0
```

Comment:
I might have misunderstood, so feel free to ignore this comment.  Should this be `0xfffffff8` instead of `0xfffffff0`?  Currently, this ensures the stack pointer is a multiple of 16, but I thought the intent was to ensure it is a multiple of eight.

Reply:
Indeed, multiple of 8 is sufficient. I confused myself by looking at OS_STACK_ALIGNMENT define in os_arch.h. That number for armv7 was inherited from x86, where that might've been a restriction set by the underlying OS (I can't remember!). And is most likely not quite right.